node/src/server-modules/GlideJson.ts

```diff
@@ -179,11 +192,154 @@ export class GlideJson {
+     * @param values - The JSON values to be inserted into the array, in JSON formatted bytes or str.
+     *     JSON string values must be wrapped with quotes. For example, to append `"foo"`, pass `"\"foo\""`.
+     * @returns
+     * - For JSONPath (path starts with `$`):
+     *       Returns an array with a list of integers for every possible path,
+     *       indicating the new length of the array, or `null` for JSON values matching
+     *       the path that are not an array. If `path` does not exist, an empty array
+     *       will be returned.
+     * - For legacy path (path doesn't start with `$`):
+     *       Returns an integer representing the new length of the array. If multiple paths are
+     *       matched, returns the length of the first modified array. If `path` doesn't
+     *       exist or the value at `path` is not an array, an error is raised.
+     * - If the index is out of bounds or `key` doesn't exist, an error is raised.
+     *
+     * @example
+     * ```typescript
+     * await GlideJson.set(client, "doc", "$", '[[], ["a"], ["a", "b"]]');
+     * const result = await GlideJson.arrinsert(client, "doc", "$[*]", 0, ['"c"', '{"key": "value"}', "true", "null", '["bar"]']);
+     * console.log(result); // Output: [5, 6, 7]
+     * const doc = await json.get(client, "doc");
+     * console.log(doc); // Output: '[["c",{"key":"value"},true,null,["bar"]],["c",{"key":"value"},true,null,["bar"],"a"],["c",{"key":"value"},true,null,["bar"],"a","b"]]'
+     * ```
+     * @example
+     * ```typescript
+     * await GlideJson.set(client, "doc", "$", '[[], ["a"], ["a", "b"]]');
+     * const result = await GlideJson.arrinsert(client, "doc", ".", 0, ['"c"'])
+     * console.log(result); // Output: 4
+     * const doc = await json.get(client, "doc");
+     * console.log(doc); // Output: '[\"c\",[],[\"a\"],[\"a\",\"b\"]]'
+     * ```
+     */
+    static async arrinsert(
+        client: BaseClient,
+        key: GlideString,
+        path: GlideString,
+        index: number,
+        values: GlideString[],
+    ): Promise<ReturnTypeJson<number>> {
+        const args = ["JSON.ARRINSERT", key, path, index.toString(), ...values];
+
+        return _executeCommand(client, args);
+    }
+
+    /**
+     * Pops an element from the array located at `path` in the JSON document stored at `key`.
+     *
+     * @param client - The client to execute the command.
+     * @param key - The key of the JSON document.
+     * @param options - (Optional) See {@link JsonArrPopOptions} and {@link DecoderOption}.
+     * @returns
+     * - For JSONPath (path starts with `$`):
+     *       Returns an array with a strings for every possible path, representing the popped JSON
+     *       values, or `null` for JSON values matching the path that are not an array
+     *       or an empty array. If a value is not an array, its corresponding return value is
+     *       `"null"`.
```

Comment:
Do we need `""`?

Reply:
Nice catch! removed from java client too